src/main/java/tp/command/AddWardCommand.java

```diff
@@ -12,48 +12,32 @@
 import tp.Ui;
 import tp.WardList;
 import tp.WardStorage;
-import tp.person.Doctor;
-import tp.person.Nurse;
-import tp.person.Patient;
 
 public class AddWardCommand extends Command {
-    protected int doctorIndex;
-    protected int patientIndex;
-    protected int nurseIndex;
+    protected int[] docIndexes;
+    protected int[] patientIndexes;
+    protected int[] nurseIndexes;
     protected int wardNumber;
 
-    public AddWardCommand() {
-    }
-
-    public AddWardCommand(int doctorIndex, int patientIndex, int nurseIndex, int wardNumber) {
-        this.doctorIndex = doctorIndex;
-        this.patientIndex = patientIndex;
-        this.nurseIndex = nurseIndex;
+    public AddWardCommand(int[] docIndexes, int[] patientIndexes, int[] nurseIndexes, int wardNumber) {
+        this.docIndexes = docIndexes;
+        this.patientIndexes = patientIndexes;
+        this.nurseIndexes = nurseIndexes;
         this.wardNumber = wardNumber;
     }
 
+
     @Override
     public String execute(DoctorList doctorList, PatientList patientList, NurseList nurseList,
                           WardList wardList, AppointmentList appointmentList, Ui ui,
                           DoctorStorage doctorStorage, WardStorage wardStorage,
                           PatientStorage patientStorage, NurseStorage nurseStorage,
                           AppointmentStorage appointmentStorage) throws IHospitalException {
-
-        if (patientIndex <= 0 || patientIndex > patientList.getSize()) {
-            throw new IHospitalException("The patient does not exist\n");
-        }
-        if (doctorIndex <= 0 || doctorIndex > doctorList.getSize()) {
-            throw new IHospitalException("The doctor does not exist\n");
-        }
-        if (nurseIndex <= 0 || nurseIndex > nurseList.getSize()) {
-            throw new IHospitalException("The nurse does not exist\n");
+        if (wardNumber <= 0 || wardNumber > wardList.getSize()) {
+            throw new IHospitalException("The ward does not exist\n");
```

Comment:
I think ward number don't need to have "non-existence" exception (since we're adding a new ward into the system), it should have exception for duplicate ID instead.